`azurerm_iothub` - support for new property `endpoint.subscription_id` (#27524)
hashicorp/terraform-provider-azurerm: website/docs/r/iothub.html.markdown

* `resource_group_name` - (Optional) The resource group in which the endpoint will be created.

* `subscription_id` - (Optional) The subscription ID for the endpoint.

> ~> **NOTE:** When `subscription_id` isn't specified it will be set to the subscription ID used in the provider block.

Author: Updated

> ~> **NOTE:** When `subscription_id` isn't specified in the tf config, it would be set with the provider's subscription ID.

Reviewer: ```suggestion
~> **NOTE:** When `subscription_id` isn't specified it will be set to the subscription ID used in the provider block.
```

Author: Updated
